src/main/java/com/pgs/client/component/Client.java

```diff
@@ -0,0 +1,22 @@
+package com.pgs.client.component;
+
+import com.pgs.client.service.AuthenticationClient;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class Client {
```

Comment:
add `@Profile("smoke-test")` to avoid having it run whenever application starts
For reference: https://www.baeldung.com/spring-profiles#use-profile-on-a-bean